### PR TITLE
Two percent sprint

### PR DESCRIPTION
**Pull request by dkirkby**
This is still work in progress...

### CODE REVIEW

File: py/desisurvey/afternoonplan.py [hunk 1]
```diff
@@ -47,108 +49,69 @@ def __init__(self, MJDstart, MJDend, surveycal, tilesubset=None):
...
+                os.environ['DESIMODEL'], 'data', 'footprint', 'desi-tiles.fits')
+        except KeyError:
+            raise RuntimeError('DESIMODEL environment variable is not defined.')
+        tiles = astropy.table.Table.read(tiles_path, hdu=1)
+
+        # Only use tables with IN_DESI set.
+        keep = tiles['IN_DESI'] > 0
+
+        # Ignore EXTRA tiles.
+        keep[tiles['PROGRAM'] == 'EXTRA'] = False
```
Comment (sbailey) [inline py/desisurvey/afternoonplan.py]: FYI, the "standard" but apparently not well known way to load `desi-tiles.fits` and trim to just tiles in the DESI footprint while discarding the PROGRAM='EXTRA' tiles:
```python
import desimodel.io
tiles = desimodel.io.load_tiles()
```
That returns an `astropy.io.fits.fitsrec.FITS_rec`; wrap with `astropy.table.Table()` if you'd prefer that.
Use keywords `onlydesi=False` and/or `extra=True` to get non-DESI and/or extra tiles.

Comment (dkirkby) [inline py/desisurvey/afternoonplan.py]: Thanks, I just made the change.

**Comment by dkirkby** [comment]
I just found a typo which explains the assertion crash @forero reported earlier today:
```
        lst13evening = mjd2lst(day_stats['MJDe13twi'])
        lst13morning = mjd2lst(day_stats['MJDe13twi'])
```
The second `MJDe13twi` should be `MJDm13twi`.  It would be great to change some of these variable names to have higher SNR to make typos like this move obvious, but that can wait until the dust settles on the current sprint.

**Comment by sbailey** [comment]
Heads up: I just pushed a small change to fix filenames for py3.

Previous code like `filename = 'obslist' + day_stats['dirName'] + '.fits'` didn't work for py3 since `day_stats['dirName']` is `numpy.bytes_` which can't be combined with the unicode strings (grumble).  The fix that works for both py2.7 and py3.5 is: `filename = 'obslist' + day_stats['dirName'].decode('ascii') + '.fits'`.

**Comment by dkirkby** [comment]
I am merging this now so I don't block other people's progress.

The main unresolved issue is moon avoidance.  I implemented an alternative to calling `avoidobject.moonLoc()` for each tile in `nextFieldSelector()` based on a pre-tabulated moon (alt,az) table stored by `nightcal.getCalAll()`.  This saves some time, but the new and old code give significantly different moon-tile separation angles and I don't know yet which (if either) is correct. I am leaving the original (slower) implementation intact and will revisit this in a new branch.

Another caveat is that I have only tested in the "simulating from zero" mode, i.e., with no `tiles_observed.fits` file from a previous run of `surveysim`.  Restarting a simulation might work, but I haven't tested it yet.
